elastic/apm-server
# Update rally cmd to use corpora generated by agent data

## Motivation/summary

Remove old `rally/gencorpora` code in favor of systemtest's `gencorpora` command which generates ES corpora based on agent data. With this PR `make rally` command can be executed to generate the required corpora and use it with rally to benchmark a local ES cluster.

The PR also removes the `cat_bulk.py` script. We don't have feature parity with the script as the script can be used to allow developers to run the `cat_bulk` server independently and send traces to it using a custom code to generate ES corpora. The feature can be added as a new command in systemtest and all the building blocks for the same are already present however, I have not added it since there is no current use case for it.

## Checklist

~~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~~
~~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~~
- [x] Documentation has been updated (verified that the document `dev_docs/rally.md` holds valid)

## How to test these changes

1. Start a local ES cluster (can use `./testing/stack-monitoring.sh`
2. Run `make rally`
3. Observe the generated rally summary to assert its success

## Related issues

Part of #8754
